openlibrary/plugins/openlibrary/js/SearchPage.js

```diff
@@ -0,0 +1,21 @@
+import { addModeInputsToForm, mode as searchMode } from './SearchUtils';
+import $ from 'jquery';
+
+/** Manages some (PROBABLY VERY FEW) of the interactions on the search page */
+export class SearchPage {
```

Comment:
Haha, that's specifically what I'm trying to avoid :P My logic being:

> I used es6 classes to organize the component logic, but that's not saying I think that's the way we should do this; it was just the easiest for me to do. Even if we decide on something else, these will make it easier to switch to that something else than what we had before.

Reply:
Does it have to be a class then...?

Reply:
Can you expand?

Reply:
All this class  does is wire up things. I'm not a big fan of classes for UI unless a framework is used and they are extending some sensible defaults.

I feel it would be more readable i it was just a plain function and we took a functional programming approach here so the input and output were clear to understand.
```
function wireupSearchPageEvents(form) {}
```

The method updateModeInputs adds unnecessary abstraction IMO.

Reply:
Closing as WONTFIX since I don't see much of  a benefit to removing classes here.